app/models.py

```diff
@@ -1039,6 +1039,10 @@ def validate(self, enforce_required=True, required_fields=None):
         ):
             errs['niceToHaveRequirements'] = 'answer_required'
 
+        if max_day_rate and 'dayRate' not in errs:
+            if float(self.data['dayRate']) > float(max_day_rate):
```

Comment:
We should fail, since we don't expect any cast errors here – there are no errors in the field after schema validation, and that checks for values matching the float format (only digits, one dot etc).